detox/src/devices/drivers/DeviceDriverBase.js

```diff
@@ -129,7 +144,7 @@ class DeviceDriverBase {
   }
 
   async cleanup(deviceId, bundleId) {
-    return await Promise.resolve('');
```

Comment:
Yes. I checked, but I can't implement it so easily because of the *coldBoot* complication.

I clearly see across the project that `.boot()` methods in *--DeviceDriver* classes are obligated to return `string` which semantically means a real device id (a concept which is contrary to device name, e.g.: `Nexus_5X_API_26 -> emulator-5556` or `iPhone 6s -> 7483CE6E-01CF-4B95-82DA-11E22AA94729`.

That would mean that if I start returning a complex object `{deviceId, coldBoot}` from a child driver, then signatures between DriverBase and ChildDriver methods will start to diverge, and I don't like this idea very much.

Also, we've got around 10-14 methods like that on DeviceDriverBase.js. It is going to pollute the class with x2 signatures count. Also, if you check the code of Device and DeviceDriverBase, you will see that the good half of methods are very dumb and repetitive there:

```
async sendToHome() {
    await this.deviceDriver.sendToHome(this._deviceId);
  async shake() {
    await this.deviceDriver.shake(this._deviceId);
  async terminateApp(bundleId) {
    const _bundleId = bundleId || this._bundleId;
    await this.deviceDriver.terminate(this._deviceId, _bundleId);
  async installApp(binaryPath) {
    const _binaryPath = binaryPath || this._binaryPath;
    await this.deviceDriver.installApp(this._deviceId, _binaryPath);
  async uninstallApp(bundleId) {
    const _bundleId = bundleId || this._bundleId;
    await this.deviceDriver.uninstallApp(this._deviceId, _bundleId);
  async reloadReactNative() { await this.deviceDriver.reloadReactNative();
```

I think that architecture in all these classes is quite ad-hocish. and can (and is worth to) be rethought and improved, but since the emitter physically lives inside DeviceDriverBase class right, I think we can postpone a heavier rewrite. I'm afraid if I really start touch it, I won't be lucky to finish it till the next week.